### PR TITLE
feat(Examples): Add 2048 Demo to MAX32655

### PR DESCRIPTION
**Pull request by sihyung-maxim**
### Description

This PR adds the 2048 Demo which splits the task of the game between the ARM and RISC-V cores for the MAX32655.

### Checklist Before Requesting Review

- [ ] PR Title follows correct guidelines.
- [ ] Description of changes and all other relevant information.
- [ ] (Optional) Link any related GitHub issues [using a keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [ ] (Optional) Provide info on any relevant functional testing/validation.  For API changes or significant features, this is not optional.

### CODE REVIEW

**Comment by sihyung-maxim** [comment]
Fixed linter errors and outdated/unfinished comments. Moved all the semaphore/mailbox definitions, for both the RISCV and ARM core, into their separate ipc_defines.h which should match.

File: Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    uint16_t readLocation;
+    uint16_t writeLocation;
+#if (MAILBOX_SIZE == 0)
+    uint8_t payload[1];
+#else
+    uint8_t payload[MAILBOX_PAYLOAD_LEN];
+#endif
+} mxcSemaBox_t;
+
+#define MAILBOX_MAIN_GRID_IDX (0) // Main grid indexes are from 0 to (16 blocks * 4 bytes) - 1.
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h:52]:
```suggestion
#define MAILBOX_MAIN_GRID_IDX (0) // Main grid indices are from 0 to (16 blocks * 4 bytes) - 1.
```

File: Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#define MAILBOX_MAIN_GRID_STATE_IDX \
+    (4 * 16) // Indexes are from (4 bytes * 16) to ((4 bytes * 16) + (1 byte * 16)))
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h:54]:
```suggestion
    (4 * 16) // Indices are from (4 bytes * 16) to ((4 bytes * 16) + (1 byte * 16)))
```

File: Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+#define MAILBOX_KEYPRESS_IDX ((4 * 16) + (1 * 16)) // All indexes before are for the main grids.
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/RISCV/inc/ipc_defines.h:55]:
```suggestion
#define MAILBOX_KEYPRESS_IDX ((4 * 16) + (1 * 16)) // All indices before are for the main grids.
```

File: Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    uint16_t readLocation;
+    uint16_t writeLocation;
+#if (MAILBOX_SIZE == 0)
+    uint8_t payload[1];
+#else
+    uint8_t payload[MAILBOX_PAYLOAD_LEN];
+#endif
+} mxcSemaBox_t;
+
+#define MAILBOX_MAIN_GRID_IDX (0) // Main grid indexes are from 0 to (16 blocks * 4 bytes) - 1.
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h:52]:
```suggestion
#define MAILBOX_MAIN_GRID_IDX (0) // Main grid indices are from 0 to (16 blocks * 4 bytes) - 1.
```

File: Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#define MAILBOX_MAIN_GRID_STATE_IDX \
+    (4 * 16) // Indexes are from (4 bytes * 16) to ((4 bytes * 16) + (1 byte * 16)))
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h:54]:
```suggestion
    (4 * 16) // Indices are from (4 bytes * 16) to ((4 bytes * 16) + (1 byte * 16)))
```

File: Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+#define MAILBOX_KEYPRESS_IDX ((4 * 16) + (1 * 16)) // All indexes before are for the main grids.
```
Comment (mharringADI) [inline Examples/MAX32655/Demo_2048/ARM/inc/ipc_defines.h:55]:
```suggestion
#define MAILBOX_KEYPRESS_IDX ((4 * 16) + (1 * 16)) // All indices before are for the main grids.
```

File: Libraries/MiscDrivers/Display/tft_ssd2119.c [hunk 7]
```diff
@@ -1197,6 +1197,351 @@ void MXC_TFT_FillRect(area_t *area, int color)
...
+    if ((area->y + h) > DISPLAY_HEIGHT) {
+        h = DISPLAY_HEIGHT - area->y;
+    }
+
+    displaySub(area->x, area->y, w, h);
+
+    int i = 0;
+    for (y = 0; y < h; y++) {
+        for (x = 0; x < w; x++) {
+            // g_fifo[0] = (0x01000100 | ((uint32_t)(image[i] & 0x00FF) << 16) | (uint32_t)((image[i] & 0xFF00) >> 8));
```
Comment (mharringADI) [inline Libraries/MiscDrivers/Display/tft_ssd2119.c:1267]: Looks like this comment can be removed?

**Review by mharringADI — COMMENTED** [review]
Works great, thanks @sihyung-maxim! A few really minor comments but functionally works as expected.

File: Libraries/MiscDrivers/Display/tft_ssd2119.c (hunk 7 above)
Comment (sihyung-maxim) [reply Libraries/MiscDrivers/Display/tft_ssd2119.c]: Yep, removed.

**Review by lorne-maxim — APPROVED** [review]

**Pull request merged**